src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzerTests.GuardedCallsTests.cs

```diff
@@ -4315,6 +4315,58 @@ void M1()
             await VerifyAnalyzerAsyncCs(source, s_msBuildPlatforms);
         }
 
+        [Fact]
+        public async Task GuardMemberWithinPlatformSpecificTypeShouldNowWarn()
+        {
+            var source = @"
+using System;
+using System.Runtime.Versioning;
+
+class Test
+{
+    void M1()
+    {
+        [|UnsupportedBrowserType.M1()|];
+        [|UnsupportedBrowserType.M2()|];
+        if (UnsupportedBrowserType.IsSupported)
```

Comment:
Handling  https://github.com/dotnet/runtime/issues/49755, when a type is annotated with platform attribute and it has a boolean member used for platform check, we can apply the guard attribute for that member so that it can be used as a guard plus we know that we don't need to warn for this member reference

Reply:
To make sure I understand the behavior here.

* This call site is reachable on all platforms, including browser
* `UnsupportedBrowserType` is unsupported on browser
* `UnsupportedBrowserType.IsSupported` is a static method that indicates whether or not `UnsupportedBrowserType` is supported on the current platform
    * Its return value would be intended to return `true` for all platforms except browser
    * That could be done either through `#ifdef` code blocks in the source, or by having `public static bool IsSupported { get => !OperatingSystem.IsBrowser(); }`
* The analyzer now recognizes that the member you're accessing is a guard member, and it respects that guard's platforms while evaluating the types referenced to reach that guard
* The result is that you can:
    * Access the guard member itself without getting any analyzer warnings
    * Access members within the guard, with the guard's assertions applying
    * And therefore not get any diagnostic messages in the scenario captured here

Is that all correct?

Reply:
Right, all correct

Reply:
Awesome; thanks! This feels like a powerful pattern we can promote. Platform-specific types can have static `IsSupported` methods/properties annotated with the guards that match the platforms the type is annotated with. Then code referencing the APIs only has to call into the `IsSupported` member, and all other guarding is taken care of.

/cc @marek-safar @terrajobst